### PR TITLE
Fixes for CMake build on Windows target

### PR DESCRIPTION
**Pull request by cjgdev**
This pull request fixes a few issues with the CMake build system on the Windows platform.

Setup:
- Windows 10 Pro Build 16299
- CMake 3.10.1
- Qt 5.10.1
- Visual Studio 2017 15.5.7

Notes:

1. It is largely considered bad practice to override the default library type when adding a library target using add_library, because the user may wish to override the default by providing -DBUILD_SHARED_LIBS=1 on the command line. You override the default (STATIC) on line 76 of src/CMakeLists.txt, however do not then export library symbols.

If the user has a version of CMake >= 3.5 then he may provide -DCMAKE_WINDOWS_EXPORT_ALL_SYMBOLS on the command-line, which will fix the linker error, but this causes ALL symbols to be exported, which results in a needlessly bloated .lib file, and requires a pretty recent version of CMake, however the minimum required version in CMakeLists.txt is 3.5.0, so perhaps this is acceptable.

tl;dr Do not override the default library type unless you minimally provide appropriate dllimport / dllexport storage-class attributes.

2. You should do **one** of either setting CMAKE_AUTOMOC=ON **OR** calling qt5_wrap_cpp, but not both. Note that the function invocation in the various CMakeLists.txt files incorrectly uses a comma to separate parameters, which is invalid CMake syntax, and causes a configuration error on Windows. I chose to remove the call to qt5_wrap_cpp, since the function invocation was wrong anyway.

4. Consider adding post build actions to deploy the necessary DLL dependencies to output folders. This is necessary on Windows to enable use of the debugger. Related to this, the windeployqt command-line utility will handle all dependencies, however for QML dependencies only a single qmldir directory may be specified, so it is useful to include all dependencies in the main .qml file. I'm sure there are other ways of improving this, but for simplicity I manually added the QtQuick.Shapes import to the root / main qml file for each sample, enabling windeployqt to find all dependencies necessary to run the binary.

After fixing these issues I was able to successfully run the samples on Windows, with the caveat that there appears to be a problem with qan::Connector::setEdgeComponent which prints the following non-descriptive error message "Error while creating edge: ()", which I have not yet tracked down to the root cause. Is this working on other platforms?

### CODE REVIEW

**Comment by cneben** [comment]
Thanks for the very detailed MR. I've no Windows box at hand until the end of the week, but your PR will be merged ASAP.

Appveyor build is still using qmake: you should try the (fully static...) qmake build with qtcreator, setEdgeComponent() error is probably related to a QML delegate (Edge.qml?) resource loaded incorrectly. 

Thanks again.

**Comment by cneben** [comment]
Still no working Windows workstation... Thanks again for your high quality PR and sorry again for the merge delay!
